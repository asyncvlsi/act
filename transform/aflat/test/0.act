
namespace bar {

bool q;

}

bool a[2], b;

prs { 
   a[1] -> b-
   a[0] -> b+
}

defproc barp (bool p, q)
{
  bool r;

  prs {
    p => r-
    q & p #> r+
  }
}

defproc foo (bool a, b)
{
  prs {
    a -> b-
  }
  barp z[6..8];
}

foo f[2];
