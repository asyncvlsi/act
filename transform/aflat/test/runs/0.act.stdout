"a[1]"->"b"-
"a[0]"->"b"+
"f[0].a"->"f[0].b"-
"f[0].z[6].p"->"f[0].z[6].r"-
~("f[0].z[6].p")->"f[0].z[6].r"+
"f[0].z[6].q"&"f[0].z[6].p"->"f[0].z[6].r"+
"f[0].z[6].q"&"f[0].z[6].p"->"f[0].z[6].r"-
"f[0].z[7].p"->"f[0].z[7].r"-
~("f[0].z[7].p")->"f[0].z[7].r"+
"f[0].z[7].q"&"f[0].z[7].p"->"f[0].z[7].r"+
"f[0].z[7].q"&"f[0].z[7].p"->"f[0].z[7].r"-
"f[0].z[8].p"->"f[0].z[8].r"-
~("f[0].z[8].p")->"f[0].z[8].r"+
"f[0].z[8].q"&"f[0].z[8].p"->"f[0].z[8].r"+
"f[0].z[8].q"&"f[0].z[8].p"->"f[0].z[8].r"-
"f[1].a"->"f[1].b"-
"f[1].z[6].p"->"f[1].z[6].r"-
~("f[1].z[6].p")->"f[1].z[6].r"+
"f[1].z[6].q"&"f[1].z[6].p"->"f[1].z[6].r"+
"f[1].z[6].q"&"f[1].z[6].p"->"f[1].z[6].r"-
"f[1].z[7].p"->"f[1].z[7].r"-
~("f[1].z[7].p")->"f[1].z[7].r"+
"f[1].z[7].q"&"f[1].z[7].p"->"f[1].z[7].r"+
"f[1].z[7].q"&"f[1].z[7].p"->"f[1].z[7].r"-
"f[1].z[8].p"->"f[1].z[8].r"-
~("f[1].z[8].p")->"f[1].z[8].r"+
"f[1].z[8].q"&"f[1].z[8].p"->"f[1].z[8].r"+
"f[1].z[8].q"&"f[1].z[8].p"->"f[1].z[8].r"-
